### PR TITLE
New functions added

### PR DESCRIPTION
**Pull request by murd0**
Added dateValidator and tableFromSql functions. Regex now rejects dates pre-1900 as suggested.

### CODE REVIEW

File: codonPython/dateValidator.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import re
+
+def validDate(string: str):
+    """Validates stringtype dates of type `dd/mm/yyyy`, `dd-mm-yyyy` or `dd.mm.yyyy` from years 1900 - 9999. Leap year support included.
+    
+    Arguments:
+        string: str
```
Comment (hadjipanteliscg) [inline codonPython/dateValidator.py]: space between `g` and `:`.

File: codonPython/dateValidator.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+            Date to be validated
+
+    Outputs:
```
Comment (hadjipanteliscg) [inline codonPython/dateValidator.py]: Returns, not Outputs.
+
`-------`

File: codonPython/tableFromSql.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
+from sqlalchemy import create_engine
+import pandas as pd
+
+def tableFromSql(server, database, table_name, user="", password="", schema=None, index_col=None, coerce_float=True, parse_dates=None, columns=None, chunksize=None):
+    '''Returns a SQL table in a DataFrame.
+    
+    Convert a table stored in SQL Server 2016 into a pandas dataframe. Requires the odbc driver for SQL Server.
+
+    Args:
```
Comment (hadjipanteliscg) [inline codonPython/tableFromSql.py]: `Parameters`
    `----------`

File: codonPython/tableFromSql.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+        server: string
```
Comment (hadjipanteliscg) [inline codonPython/tableFromSql.py]: space between `r` and `:`

File: codonPython/tableFromSql.py [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+            Name of SQL table in database.
+
+        schema : string, default None
+            Name of SQL schema in database to query (if database flavor supports this). Uses default schema if None (default).
+        
+        index_col : string or list of strings, optional, default: None
+            Column(s) to set as index(MultiIndex).
+
+        coerce_float : boolean, default True
+            Attempts to convert values of non-string, non-numeric objects (like decimal.Decimal) to floating point. Can result in loss of Precision.
```
Comment (hadjipanteliscg) [inline codonPython/tableFromSql.py]: Too long. New line.

File: codonPython/tableFromSql.py [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+        
+        parse_dates : list or dict, default: None
+            - List of column names to parse as dates.
+            - Dict of {column_name: format string} where format string is strftime compatible in case of parsing string times or is one of (D, s, ns, ms, us) in case of parsing integer timestamps.
```
Comment (hadjipanteliscg) [inline codonPython/tableFromSql.py]: Too long. New line.

File: codonPython/tableFromSql.py [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+            - Dict of {column_name: arg dict}, where the arg dict corresponds to the keyword arguments of pandas.to_datetime() Especially useful with databases without native Datetime support, such as SQLite.
```
Comment (hadjipanteliscg) [inline codonPython/tableFromSql.py]: Too long. New line.

File: codonPython/tableFromSql.py [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+        chunksize : int, default None
+            If specified, returns an iterator where chunksize is the number of rows to include in each chunk.
+
+        	
+    Returns:
+        pd.DataFrame
+            Dataframe of the table requested from sql server
+
+    >>> tableFromSql("myServer2", "myDatabase2", "myTable2")
+    pd.DataFrame
```
Comment (hadjipanteliscg) [inline codonPython/tableFromSql.py]: This will fail. Comment out.

File: codonPython/tableFromSql.py [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+
+    >>> tableFromSql("myServer", "myDatabase", "myTable", schema="specialSchema", columns=["col_1", "col_3"])
+    pd.DataFrame
```
Comment (hadjipanteliscg) [inline codonPython/tableFromSql.py]: This will fail. Comment out.

File: codonPython/tableFromSql.py [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+    '''
+
+    uri = "mssql+pyodbc://{}:{}@{}/{}?driver=SQL Server Native Client 11.0".format(user, password, server, database)
+    engine = create_engine(uri)
+
+    return pd.read_sql_table(table_name, engine, schema=None, index_col=None, coerce_float=True, parse_dates=None, columns=None, chunksize=None)
```
Comment (hadjipanteliscg) [inline codonPython/tableFromSql.py]: try/except /else this.

File: codonPython/tableFromSql.py [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+
+    >>> tableFromSql("myServer2", "myDatabase2", "myTable2")
+    pd.DataFrame
+
+    >>> tableFromSql("myServer", "myDatabase", "myTable", schema="specialSchema", columns=["col_1", "col_3"])
+    pd.DataFrame
+    '''
+
+    uri = "mssql+pyodbc://{}:{}@{}/{}?driver=SQL Server Native Client 11.0".format(user, password, server, database)
+    engine = create_engine(uri)
```
Comment (hadjipanteliscg) [inline codonPython/tableFromSql.py]: try/except /else this.

File: codonPython/tableFromSql.py [hunk 12]
```diff
@@ -0,0 +1,60 @@
...
+            Attempts to convert values of non-string, non-numeric objects (like decimal.Decimal) to floating point. Can result in loss of Precision.
+        
+        parse_dates : list or dict, default: None
+            - List of column names to parse as dates.
+            - Dict of {column_name: format string} where format string is strftime compatible in case of parsing string times or is one of (D, s, ns, ms, us) in case of parsing integer timestamps.
+            - Dict of {column_name: arg dict}, where the arg dict corresponds to the keyword arguments of pandas.to_datetime() Especially useful with databases without native Datetime support, such as SQLite.
+
+        columns : list, default: None
+            List of column names to select from SQL table
+
```
Comment (hadjipanteliscg) [inline codonPython/tableFromSql.py]: delete line.

File: codonPython/tableFromSql.py [hunk 13]
```diff
@@ -0,0 +1,60 @@
...
+            Column(s) to set as index(MultiIndex).
+
+        coerce_float : boolean, default True
+            Attempts to convert values of non-string, non-numeric objects (like decimal.Decimal) to floating point. Can result in loss of Precision.
+        
+        parse_dates : list or dict, default: None
+            - List of column names to parse as dates.
+            - Dict of {column_name: format string} where format string is strftime compatible in case of parsing string times or is one of (D, s, ns, ms, us) in case of parsing integer timestamps.
+            - Dict of {column_name: arg dict}, where the arg dict corresponds to the keyword arguments of pandas.to_datetime() Especially useful with databases without native Datetime support, such as SQLite.
+
```
Comment (hadjipanteliscg) [inline codonPython/tableFromSql.py]: delete line

File: codonPython/tableFromSql.py [hunk 14]
```diff
@@ -0,0 +1,60 @@
...
+        user: string, default: ""
+            If verification is required, name of the user
+
+        password: string, default: ""
+            If verification is required, password of the user
+
+        table_name : string
+            Name of SQL table in database.
+
```
Comment (hadjipanteliscg) [inline codonPython/tableFromSql.py]: delete line.

File: codonPython/dateValidator.py [hunk 15]
```diff
@@ -0,0 +1,26 @@
...
+        boolean
+            Whether the date is valid or not
+
+    >>> validDate("11/02/1996")
+    True
+
```
Comment (hadjipanteliscg) [inline codonPython/dateValidator.py]: delete line.

File: codonPython/dateValidator.py [hunk 16]
```diff
@@ -0,0 +1,26 @@
+import re
+
+def validDate(string: str):
+    """Validates stringtype dates of type `dd/mm/yyyy`, `dd-mm-yyyy` or `dd.mm.yyyy` from years 1900 - 9999. Leap year support included.
+    
+    Arguments:
```
Comment (hadjipanteliscg) [inline codonPython/dateValidator.py]: Change to     
`Parameters`
 `----------`

File: codonPython/dateValidator.py [hunk 17]
```diff
@@ -0,0 +1,26 @@
+import re
```
[the remark this answers is not in the document]
Comment (murd0) [reply codonPython/dateValidator.py]: So this is necessary even for default libraries?

[the remark this answers is not in the document]
Comment (hadjipanteliscg) [reply codonPython/dateValidator.py]: No. Ignore me.

**Pull request closed without merging**